File: Backend/Models/Device.cs
using Backend.Models.Enums;

namespace Backend.Models;

public class Device
{
    public int Id { get; set; }

    public DeviceType Type { get; set; }

    public double Latitude { get; set; }
    public double Longitude { get; set; }

    public double? OrientationAngle { get; set; }

    public bool IsActive { get; set; }

    public int AreaId { get; set; }
    public Area Area { get; set; } = null!;
    public int PlaceId { get; set; }
    public Place Place { get; set; } = null!;

    public int? TargetId { get; set; }
    public Target? Target { get; set; }

    public ICollection<DeviceUser> DeviceUsers { get; set; } = new List<DeviceUser>();
}

File: Frontend/src/models/Device.ts
export interface Device {
  id: number;
  type: "Camera" | "Radar";

  placeId: number;
  areaId: number;

  latitude: number;
  longitude: number;

  isActive: boolean;

  orientationAngle?: number | null;

  targetId?: number | null;
  targetName?: string | null;


  targetLatitude?: number | null;
  targetLongitude?: number | null;

    userIds?: string[];
}
